users/forms.py

```diff
@@ -24,3 +25,18 @@ class Meta:
         model = CustomUser
         fields = '__all__'
         widgets = {'country': CountrySelectWidget()}
+
+
+class LoginAuthentication(AuthenticationForm):
+    def clean(self):
+        username = self.cleaned_data.get('username')
+        if "@" not in username:
+            username = username + "@codepoets.it"
+        password = self.cleaned_data.get('password')
+
+        if username is not None and password:
+            self.user_cache = authenticate(self.request, username=username, password=password)
+            if self.user_cache is None:
+                raise self.get_invalid_login_error()
+            else:
+                self.confirm_login_allowed(self.user_cache)
```

Comment:
Each `clean` method in Django form must return data. I think it is enough to return `self.cleaned_data` here.